src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs

```diff
@@ -50,83 +52,172 @@ static IdentityModelEventSource()
+            }
+        }
+
         [Event(5, Level = EventLevel.Critical)]
         public void WriteCritical(string message)
         {
-            if (IsEnabled() && _logLevel >= EventLevel.Critical)
+            if (IsEnabled() && LogLevel >= EventLevel.Critical)
             {
-                message = PrepareMessage(message, EventLevel.Critical);
+                message = PrepareMessage(EventLevel.Critical, message);
                 WriteEvent(5, message);
             }
         }
 
         [NonEvent]
-        public void Write(EventLevel level, string message, Exception innerException)
+        public void WriteCritical(string message, params object[] args)
+        {
+            if (IsEnabled() && LogLevel >= EventLevel.Critical)
+            {
+                if (args != null)
+                    WriteCritical(string.Format(CultureInfo.InvariantCulture, message, args));
+                else
+                    WriteCritical(message);
+            }
+        }
+
+        [NonEvent]
+        public void Write(EventLevel level, Exception innerException, string message)
+        {
+            Write(level, innerException, message, null);
+        }
+
+        [NonEvent]
+        public void Write(EventLevel level, Exception innerException, string message, params object[] args)
         {
             if (innerException != null)
             {
-                message = String.Format(CultureInfo.InvariantCulture, "Message: {0}, InnerException: {1}", message, innerException.ToString());
+                message = string.Format(CultureInfo.InvariantCulture, "Message: {0}, InnerException: {1}", message, innerException.Message);
             }
 
             switch (level)
             {
+                case EventLevel.LogAlways:
+                    WriteAlways(message, args);
+                    break;
                 case EventLevel.Critical:
-                    WriteCritical(message);
+                    WriteCritical(message, args);
                     break;
                 case EventLevel.Error:
-                    WriteError(message);
+                    WriteError(message, args);
                     break;
                 case EventLevel.Warning:
-                    WriteWarning(message);
+                    WriteWarning(message, args);
                     break;
                 case EventLevel.Informational:
-                    WriteInformation(message);
+                    WriteInformation(message, args);
                     break;
                 case EventLevel.Verbose:
-                    WriteVerbose(message);
+                    WriteVerbose(message, args);
                     break;
                 default:
-                    WriteError(string.Format(CultureInfo.InvariantCulture, LogMessages.MIML11002, level.ToString()));
+                    WriteError(string.Format(CultureInfo.InvariantCulture, LogMessages.MIML11002, level));
+                    WriteError(message, args);
```

Comment:
I understand that and it makes sense to log that particular information - "log level unknown" as error. But logging the associate message as an error might be too much.